docs/coding_style.md

```diff
@@ -0,0 +1,190 @@
+# Coding Conventions
+
+
+## Background
+
+To improve the quality and maintainability of the code, we summarized some common coding standards and conventions.
+
+There are many style guides, and they may conflict with each other. To avoid overly arguing formatting, we make decisions based on the following priorities:
+
+- [Google Python Style](https://google.github.io/styleguide/pyguide.html#s3.8-comments-and-docstrings), [PEP 8](https://peps.python.org/pep-0008/)
+- Framework Style
+- Internal Style
+- Sub-module specific Style
+
+## Rules
+
+> Note: The sub-tile naming is following [Google Python Style](https://google.github.io/styleguide/pyguide.html#s3.8-comments-and-docstrings) and [PEP 8](https://peps.python.org/pep-0008/). See the relevant section for more details.
+
+
+### Imports
+
+- Recommend
+
+```python
+import os
+import sys
+
+from x import py
+from x import y as z
+from copy import deepcopy
+from subprocess import Popen, PIPE
```

Comment:
```suggestion
import subprocess
```

Import modules only https://google.github.io/styleguide/pyguide.html#224-decision

Reply:
This means we recommend to only import the module used by the code rather than import the whole package.

Reply:
Popen and PIPE are classes / constants in the module but not modules themselves. Double checking since this is inconsistent with the style guide.